File: recognition/app/recognizer/Producer.py
import json

from kafka import KafkaProducer

from app.utils.env import KAFKA_HOST, KAFKA_PORT


class Producer:

    def __init__(self):
        self._producer = KafkaProducer(bootstrap_servers=f"{KAFKA_HOST}:{KAFKA_PORT}")

    def produce(self, topic: str, recognized_object: str):
        self._producer.send(
            topic, json.dumps({"value": recognized_object}).encode("utf-8")
        )

File: docs/ds-report/code/producer.ts
const kafka: Kafka = new Kafka({
  clientId: `SENSOR_${SENSOR_CODE}`,
  brokers: [`${kafkaHost}:${kafkaPort}`]
})

export const produce = async (): Promise<void> => {
  const producer: Producer = kafka.producer({ createPartitioner: Partitioners.LegacyPartitioner })
  await producer.connect()
  setInterval(async (): Promise<void> => {
    await producer.send({
      topic: `SENSOR_${sourceSensor.deviceId.code}`,
      messages: [
        {
          value: JSON.stringify(values)
        }
      ]
    })
  }, sourceSensor.intervalMillis)
}
